teeio-validator/library/helperlib/pcap.c

```diff
@@ -0,0 +1,115 @@
+/**
+ *  Copyright Notice:
+ *  Copyright 2021-2022 DMTF. All rights reserved.
+ *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/spdm-emu/blob/main/LICENSE.md
+ **/
+
+#include <stdint.h>
+#include <stdio.h>
+#include <time.h>
+#include <stdbool.h>
+#include <industry_standard/pcap.h>
+#include <industry_standard/link_type_ex.h>
+#include "pcap.h"
+#include "command.h"
+
+#define PCAP_PACKET_MAX_SIZE 0x00010000
+
+static FILE *m_pcap_file;
+
+bool open_pcap_packet_file(const char *pcap_file_name, uint32_t transport_layer)
+{
+    pcap_global_header_t pcap_global_header;
+
+    if (pcap_file_name == NULL) {
+        return false;
+    }
+
+    pcap_global_header.magic_number = PCAP_GLOBAL_HEADER_MAGIC;
+    pcap_global_header.version_major = PCAP_GLOBAL_HEADER_VERSION_MAJOR;
+    pcap_global_header.version_minor = PCAP_GLOBAL_HEADER_VERSION_MINOR;
+    pcap_global_header.this_zone = 0;
+    pcap_global_header.sig_figs = 0;
+    pcap_global_header.snap_len = PCAP_PACKET_MAX_SIZE;
+    if (transport_layer == SOCKET_TRANSPORT_TYPE_MCTP) {
+        pcap_global_header.network = LINKTYPE_MCTP;
+    } else if (transport_layer == SOCKET_TRANSPORT_TYPE_PCI_DOE) {
+        pcap_global_header.network = LINKTYPE_PCI_DOE;
+    } else {
+        return false;
+    }
+
+    if ((m_pcap_file = fopen(pcap_file_name, "wb")) == NULL) {
+        printf("!!!Unable to open pcap file %s!!!\n", pcap_file_name);
+        return false;
+    }
+
+    if ((fwrite(&pcap_global_header, 1, sizeof(pcap_global_header),
+                m_pcap_file)) != sizeof(pcap_global_header)) {
+        printf("!!!Write pcap file error!!!\n");
```

Comment:
Please use TEEIO_DEBUG to print log information.

Reply:
done